pkg/inngest/inngest/_internal/client_lib/client.py

```diff
@@ -453,15 +474,30 @@ async def send(
         if isinstance(req, Exception):
             raise req
 
-        result = models.SendEventsResult.from_raw(
-            (
-                await net.fetch_with_thready_safety(
-                    self._http_client,
-                    self._http_client_sync,
-                    req,
-                )
-            ).json()
-        )
+        resp = None
+        for attempt in range(MAX_RETRY_ATTEMPTS):
+            resp = await net.fetch_with_thready_safety(
+                self._http_client,
+                self._http_client_sync,
+                req,
+            )
+            if resp.status_code < 400:
```

Comment:
```suggestion
            if resp.status_code < 500:
```

We should also break on `4xx` errors. We're retrying the same request every time so we'd keep hitting the same status code on subsequent attempts (e.g. event key is wrong)

Reply:
yeah I read and understood the comment in inngestgo but somehow copied it wrong here 😬 . Maybe overkill but I added a test that 400s don't retry